lib/src/storage_impl.dart

```diff
@@ -284,7 +284,7 @@ class _BucketImpl implements Bucket {
       throw ArgumentError('Content-Type is required for update');
     }
     md._object.acl ??= _defaultObjectAcl._toObjectAccessControlList();
-    return _api.objects.update(object, bucketName, objectName);
+    return _api.objects.patch(object, bucketName, objectName);
```

Comment:
What is the difference? And why are those semantics expected?

(Note, I'm not saying this is will documented, just that we risk breaking users).

Reply:
The `update` call uses a PUT behind the scenes, which _completely_ replaces the metadata on the object - it's not really an update, but an _overwrite_. In the other language implementations, `updateMetadata()` takes a potentially partial metadata map, and applies those changes that are present, not so in this library.

PATCH, on the other hand, allows for exactly that, bringing the Dart library in line with all other implementations.

Reply:
If  we change the semantics of an API people might be relying on we have to do a major version break.
I think it's better to fix this by documenting the overwriting behavior.

Reply:
The REST API actually uses the verb `update` too:
https://cloud.google.com/storage/docs/json_api/v1/objects/update

And then clarifies:
>Note: This method sets the complete metadata of an object. You should generally use the PATCH method instead, which changes only metadata you define in the request, while leaving other metadata unaffected.

I don't disagree that the choice of words here is unfortunate. Honestly, I think this is better fixed by creating a new package that provides a better documentation wrapper for Google Cloud Storage.

Reply:
Technically it's breaking for all the people who implement the API in-order to mock it :/

But it might be acceptable.. That said I think you would be better of forking this code and create a new  `gcs` package or something like that. A package that exclusively focuses on Google Cloud Storage. I have limited review bandwidth for this package, and is constantly concerned that side effects might break other users.